src/test/unitTests.js

```diff
@@ -0,0 +1,139 @@
+const { describe, it, beforeEach, afterEach } = require('mocha')
+const axios = require('axios')
+const chai = require('chai')
+const sinon = require('sinon')
+const UserDataHandler = require('../data_handlers/user_data_handler')
+
+let sandbox
+let userDataHandler
+const expect = chai.expect
+
+describe('UserDataHandler', () => {
+  beforeEach(() => {
+    sandbox = sinon.createSandbox()
+    userDataHandler = new UserDataHandler()
+  })
+
+  afterEach(() => {
+    sandbox.restore()
+  })
+
+  it('should load user data', async () => {
+    const data = [{ id: 1, email: 'test1@gmail.com' }, { id: 2, email: 'test2@gmail.com' }]
+    sandbox.stub(axios, 'get').returns(Promise.resolve({ data }))
+
+    await userDataHandler.loadUsers()
+    expect(userDataHandler.users.length).to.equal(2)
+  })
+
+  it('should return user emails list', () => {
+    userDataHandler.users = [{ id: 1, email: 'test1@gmail.com' }, { id: 2, email: 'test2@gmail.com' }]
+
+    const emails = userDataHandler.getUserEmailsList()
+    expect(emails).to.equal('test1@gmail.com;test2@gmail.com')
+  })
+
+  it('should return number of users', () => {
+    userDataHandler.users = [{ id: 1, email: 'test1@gmail.com' }, { id: 2, email: 'test2@gmail.com' }]
+
+    const count = userDataHandler.getNumberOfUsers()
+    expect(count).to.equal(2)
+  })
+
+  it('should return matching users', () => {
+    userDataHandler.users = [
+      { id: 1, email: 'test1@gmail.com', name: 'Test 1' },
+      { id: 2, email: 'test2@gmail.com', name: 'Test 2' }
+    ]
+
+    const users = userDataHandler.findUsers({ name: 'Test 1' })
+    expect(users.length).to.equal(1)
+    expect(users[0].email).to.equal('test1@gmail.com')
+  })
+  it('should get the number of users', () => {
+    userDataHandler.users = [{ email: 'test1@test.com' }, { email: 'test2@test.com' }]
+    const userCount = userDataHandler.getNumberOfUsers()
+    expect(userCount).to.equal(2)
+  })
+
+  it('should handle no users while getting number of users', () => {
+    userDataHandler.users = []
+    const userCount = userDataHandler.getNumberOfUsers()
+    expect(userCount).to.equal(0)
+  })
+
+  it('should find matching users ', () => {
+    userDataHandler.users = [{ name: 'John', age: 30 }, { name: 'Tom', age: 35 }]
+    const searchParams = { age: 30 }
+    const matchingUsers = userDataHandler.findUsers(searchParams)
+    expect(matchingUsers).to.deep.equal([{ name: 'John', age: 30 }])
+  })
+
+  it('should handle no search parameters while finding users', () => {
+    userDataHandler.users = [{ name: 'John', age: 30 }, { name: 'Tom', age: 35 }]
+    try {
+      userDataHandler.findUsers('')
+    } catch (err) {
+      expect(err.message).to.equal('No search parameters provided!')
```

Comment:
if the method findUsers() does not throw any exception, the test will pass. use expect.toThrow():
https://jestjs.io/docs/expect#tothrowerror